### PR TITLE
Build time configurable library management

### PR DESCRIPTION
**Pull request by andreaTP**
- [X] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

This is a pre-requisite to automate integration tests for alternative implementations of `librarymanagement` as discussed in the Gitter channel with @eed3si9n and @leonardehrenfried

### CODE REVIEW

**Comment by andreaTP** [comment]
ping for a review, this should be easy.

File: project/Dependencies.scala [hunk 1]
```diff
@@ -10,7 +10,11 @@ object Dependencies {
   // sbt modules
   private val ioVersion = "1.2.1"
   private val utilVersion = "1.2.2"
-  private val lmVersion = "1.2.1"
+  private val lmVersion =
+    sys.props.get("sbt.librarymanagement.version") match {
```
Comment (eed3si9n) [inline project/Dependencies.scala]: I recently added build-time flag for fatal warnings and I used `sbt.build` prefix.
Could we rename this to `sbt.build.lm.version`?

File: project/Dependencies.scala [hunk 2]
```diff
@@ -24,7 +28,22 @@ object Dependencies {
   private val utilScripted = "org.scala-sbt" %% "util-scripted" % utilVersion
 
   private val libraryManagementCore = "org.scala-sbt" %% "librarymanagement-core" % lmVersion
-  private val libraryManagementIvy = "org.scala-sbt" %% "librarymanagement-ivy" % lmVersion
+
+  private val libraryManagementImpl = {
+    val lmGroupID =
+      sys.props.get("sbt.librarymanagement.groupID") match {
```
Comment (eed3si9n) [inline project/Dependencies.scala]: `groupID` is a Maven term. My preference is `sbt.build.lm.organization`.

File: project/Dependencies.scala [hunk 3]
```diff
@@ -24,7 +28,22 @@ object Dependencies {
...
+        case Some(impl) => impl
+        case _ => "org.scala-sbt"
+      }
+
+    val lmArtifactID =
+      sys.props.get("sbt.librarymanagement.artifactID") match {
```
Comment (eed3si9n) [inline project/Dependencies.scala]: Similarly `sbt.build.lm.moduleName`.

**Review by eed3si9n — CHANGES_REQUESTED** [review]
Some minor comments.
Thanks for pushing this forward.

**Comment by andreaTP** [comment]
Done all requested renamings, squashed and rebased on latest develop.

**Review by eed3si9n — APPROVED** [review]
LGTM pending CI

**Pull request merged**